### DISCUSSION TITLE
if_updatenr config in wayland/sway always crashes after a couple hours or so

### DISCUSSION THREAD

**Discussion by CaeriTech** [brndnmtthws/conky#1933] [Q&A]
This conky run in wayland/sway crashes with the error shown below.  Is there something I need to change in the config to avoid this?

The config...
-----------------
conky.config = {
use_xft = true,
font = "DejaVuSans :size=12:",
default_color = "E4E4E4",
default_shade_color = "E4E4E4",
default_outline_color = "E4E4E4",
xftalpha = 1,
draw_shades = false,
draw_borders = false,
own_window = true,
own_window_type = "override",
own_window_transparent = true,
own_window_argb_visual = true,
own_window_argb_value = 155,
own_window_hints = "undecorated,below,sticky,skip_taskbar,skip_pager,below",
double_buffer = true,
text_buffer_size = 9000,
max_user_text = 22000,
alignment = "top_left",
gap_x = 5,
gap_y = 5,
max_text_width = 100,
minimum_width = 190,
maximum_width = 190,
no_buffers = true,
net_avg_samples = 10,
update_interval = 1
}
conky.text = [[
#---------------------------------------------------------------------------------------------
${alignr}${font :size=9:style=bold:}${color}_/processes
${voffset -005}\
\
${font :style=normal:size=7:}${color}\
\
${if_updatenr 1}\
\
${goto 087}cpu${goto 121}mem${goto 160}time
${top name 1}${goto 080}${font :style=bold:size=7:}${top cpu 1}${font :style=normal:size=7:}${goto 115}${top mem 1}${alignr}${top time 1}
${top name 2}${goto 080}${font :style=bold:size=7:}${top cpu 2}${font :style=normal:size=7:}${goto 115}${top mem 2}${alignr}${top time 2}
${top name 3}${goto 080}${font :style=bold:size=7:}${top cpu 3}${font :style=normal:size=7:}${goto 115}${top mem 3}${alignr}${top time 3}
${top name 4}${goto 080}${font :style=bold:size=7:}${top cpu 4}${font :style=normal:size=7:}${goto 115}${top mem 4}${alignr}${top time 4}
${top name 5}${goto 080}${font :style=bold:size=7:}${top cpu 5}${font :style=normal:size=7:}${goto 115}${top mem 5}${alignr}${top time 5}
${voffset -003}\
\
${endif}\
\
${if_updatenr 2}\
\
${goto 087}cpu${goto 121}mem${goto 160}time
${top_mem name 1}${goto 080}${top_mem cpu 1}${goto 115}${font :style=bold:size=7:}${top_mem mem 1}${font :style=normal:size=7:}${alignr}${top_mem time 1}
${top_mem name 2}${goto 080}${top_mem cpu 2}${goto 115}${font :style=bold:size=7:}${top_mem mem 2}${font :style=normal:size=7:}${alignr}${top_mem time 2}
${top_mem name 3}${goto 080}${top_mem cpu 3}${goto 115}${font :style=bold:size=7:}${top_mem mem 3}${font :style=normal:size=7:}${alignr}${top_mem time 3}
${top_mem name 4}${goto 080}${top_mem cpu 4}${goto 115}${font :style=bold:size=7:}${top_mem mem 4}${font :style=normal:size=7:}${alignr}${top_mem time 4}
${top_mem name 5}${goto 080}${top_mem cpu 5}${goto 115}${font :style=bold:size=7:}${top_mem mem 5}${font :style=normal:size=7:}${alignr}${top_mem time 5}
${voffset -003}\
\
${endif}\
\
${if_updatenr 3}\
\
${goto 087}cpu${goto 121}mem${goto 160}time
${top_time name 1}${goto 080}${top_time cpu 1}${goto 115}${top_time mem 1}${alignr}${font :style=bold:size=7:}${top_time time 1}${font :style=normal:size=7:}
${top_time name 2}${goto 080}${top_time cpu 2}${goto 115}${top_time mem 2}${alignr}${font :style=bold:size=7:}${top_time time 2}${font :style=normal:size=7:}
${top_time name 3}${goto 080}${top_time cpu 3}${goto 115}${top_time mem 3}${alignr}${font :style=bold:size=7:}${top_time time 3}${font :style=normal:size=7:}
${top_time name 4}${goto 080}${top_time cpu 4}${goto 115}${top_time mem 4}${alignr}${font :style=bold:size=7:}${top_time time 4}${font :style=normal:size=7:}
${top_time name 5}${goto 080}${top_time cpu 5}${goto 115}${top_time mem 5}${alignr}${font :style=bold:size=7:}${top_time time 5}${font :style=normal:size=7:}
${voffset -003}\
\
${endif}\
\
${if_updatenr 4}\
\
${goto 087}cpu${goto 121}mem${goto 160}time
${top_mem name 1}${goto 080}${top_mem cpu 1}${goto 115}${font :style=bold:size=7:}${top_mem mem 1}${font :style=normal:size=7:}${alignr}${top_mem time 1}
${top_mem name 2}${goto 080}${top_mem cpu 2}${goto 115}${font :style=bold:size=7:}${top_mem mem 2}${font :style=normal:size=7:}${alignr}${top_mem time 2}
${top_mem name 3}${goto 080}${top_mem cpu 3}${goto 115}${font :style=bold:size=7:}${top_mem mem 3}${font :style=normal:size=7:}${alignr}${top_mem time 3}
${top_mem name 4}${goto 080}${top_mem cpu 4}${goto 115}${font :style=bold:size=7:}${top_mem mem 4}${font :style=normal:size=7:}${alignr}${top_mem time 4}
${top_mem name 5}${goto 080}${top_mem cpu 5}${goto 115}${font :style=bold:size=7:}${top_mem mem 5}${font :style=normal:size=7:}${alignr}${top_mem time 5}
${voffset -003}\
\
${endif}
]]

The startup...
-------------------
conky: desktop window (0x310) is root window
conky: window type - override
conky: drawing to created window (0x1200002)
conky: drawing to double buffer
conky: one or more $endif's are missing
conky: FOUND: console
conky: FOUND: ncurses
conky: FOUND: file
conky: FOUND: x11
conky: FOUND: wayland

The error...
---------------
conky: /usr/src/debug/conky/conky/src/fonts.cc:89: int font_height(): Assertion `selected_font < fonts.size()' failed.
zsh: IOT instruction (core dumped)  conky -c /run/media/caeri/development/conky\~conf\~updatenr

My software versions...
--------------------------------
conky 1.21.1-1
libx11 1.8.9-1
wayland 1.22.0-1
sway 1:1.9-3

**Comment by Caellian** [comment]
Moved to issue #1935.
